src/shogun/base/SGObject.cpp

```diff
@@ -741,3 +754,14 @@ CSGObject* CSGObject::clone()
 	SG_DEBUG("leaving %s::clone(): Clone successful\n", get_name());
 	return copy;
 }
+
+void CSGObject::set(int tag_id, const Any& any)
```

Comment:
@lisitsyn That is what I wanted to confirm. It calls fire `fire_updated(tag_id);` which is a virtual function. What is the purpose of `fire_update()`. I have taken this from aer.